picocom.c

```diff
@@ -1539,11 +1542,15 @@ loop(void)
         if ( FD_ISSET(tty_fd, &wrset) ) {
 
             /* write to port */
-
             int sz;
             sz = (tty_q.len < tty_write_sz) ? tty_q.len : tty_write_sz;
             do {
-                n = write(tty_fd, tty_q.buff, sz);
+                if (tty_sleep.tv_sec | tty_sleep.tv_nsec) {
+                    nanosleep(&tty_sleep, NULL);
+                    n = write(tty_fd, tty_q.buff, 1);
```

Comment:
I wonder if we shouldn't add a for-loop here, iterating until 'sz' is met to save some overhead? Returning with n=1 here and let the mainloop do the work seems a bit pricey to me.